
vop2_win_data rk3568_vop_win_data[] reordering to:

Esmart1-win0
Esmart0-win0
Smart0-win0
Smart1-win0
Cluster0-win0
Cluster1-win0

This makes Kodi and MythTV working out-of-box with EGL DMAbuf and DRM planes HW decoded video rendering :-)


diff --speed-large-files --no-dereference --minimal -Naur linux-6.7-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c linux-6.7-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
--- linux-6.7-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c	2023-12-18 00:19:28.000000000 +0100
+++ linux-6.7-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c	2023-12-27 16:27:20.263415754 +0100
@@ -150,52 +150,52 @@
  */
 static const struct vop2_win_data rk3568_vop_win_data[] = {
 	{
-		.name = "Smart0-win0",
-		.phys_id = ROCKCHIP_VOP2_SMART0,
-		.base = 0x1c00,
-		.formats = formats_smart,
-		.nformats = ARRAY_SIZE(formats_smart),
+		.name = "Esmart1-win0",
+		.phys_id = ROCKCHIP_VOP2_ESMART1,
+		.formats = formats_rk356x_esmart,
+		.nformats = ARRAY_SIZE(formats_rk356x_esmart),
 		.format_modifiers = format_modifiers,
-		.layer_sel_id = 3,
+		.base = 0x1a00,
+		.layer_sel_id = 6,
 		.supported_rotations = DRM_MODE_REFLECT_Y,
 		.type = DRM_PLANE_TYPE_PRIMARY,
 		.max_upscale_factor = 8,
 		.max_downscale_factor = 8,
 		.dly = { 20, 47, 41 },
 	}, {
-		.name = "Smart1-win0",
-		.phys_id = ROCKCHIP_VOP2_SMART1,
-		.formats = formats_smart,
-		.nformats = ARRAY_SIZE(formats_smart),
+		.name = "Esmart0-win0",
+		.phys_id = ROCKCHIP_VOP2_ESMART0,
+		.formats = formats_rk356x_esmart,
+		.nformats = ARRAY_SIZE(formats_rk356x_esmart),
 		.format_modifiers = format_modifiers,
-		.base = 0x1e00,
-		.layer_sel_id = 7,
+		.base = 0x1800,
+		.layer_sel_id = 2,
 		.supported_rotations = DRM_MODE_REFLECT_Y,
 		.type = DRM_PLANE_TYPE_PRIMARY,
 		.max_upscale_factor = 8,
 		.max_downscale_factor = 8,
 		.dly = { 20, 47, 41 },
 	}, {
-		.name = "Esmart1-win0",
-		.phys_id = ROCKCHIP_VOP2_ESMART1,
-		.formats = formats_rk356x_esmart,
-		.nformats = ARRAY_SIZE(formats_rk356x_esmart),
+		.name = "Smart0-win0",
+		.phys_id = ROCKCHIP_VOP2_SMART0,
+		.base = 0x1c00,
+		.formats = formats_smart,
+		.nformats = ARRAY_SIZE(formats_smart),
 		.format_modifiers = format_modifiers,
-		.base = 0x1a00,
-		.layer_sel_id = 6,
+		.layer_sel_id = 3,
 		.supported_rotations = DRM_MODE_REFLECT_Y,
 		.type = DRM_PLANE_TYPE_PRIMARY,
 		.max_upscale_factor = 8,
 		.max_downscale_factor = 8,
 		.dly = { 20, 47, 41 },
 	}, {
-		.name = "Esmart0-win0",
-		.phys_id = ROCKCHIP_VOP2_ESMART0,
-		.formats = formats_rk356x_esmart,
-		.nformats = ARRAY_SIZE(formats_rk356x_esmart),
+		.name = "Smart1-win0",
+		.phys_id = ROCKCHIP_VOP2_SMART1,
+		.formats = formats_smart,
+		.nformats = ARRAY_SIZE(formats_smart),
 		.format_modifiers = format_modifiers,
-		.base = 0x1800,
-		.layer_sel_id = 2,
+		.base = 0x1e00,
+		.layer_sel_id = 7,
 		.supported_rotations = DRM_MODE_REFLECT_Y,
 		.type = DRM_PLANE_TYPE_PRIMARY,
 		.max_upscale_factor = 8,
